mne_bids/copyfiles.py

```diff
@@ -564,6 +564,15 @@ def copyfile_eeglab(src, dest):
         new_value[0, 0] = np.atleast_1d(np.array(tail))
         eeg['data'] = new_value
 
+        # Make sure all values are saved as float
+        # (see https://github.com/mne-tools/mne-bids/issues/1122)
+        eeg['nbchan'] = float(eeg['nbchan'])
+        eeg['pnts'] = float(eeg['pnts'])
+        eeg['trials'] = float(eeg['trials'])
+        eeg['srate'] = float(eeg['srate'])
+        eeg['xmin'] = float(eeg['xmin'])
+        eeg['xmax'] = float(eeg['xmax'])
```

Comment:
I fear this is not a future-proof solution ... the only thing we should be modifying is the pointer to the filename and leaving everything else unchanged

Reply:
@jasmainak Very good point. Your comment made me think of another solution. Passing `mat_dtype=True` to `loadmat` also fixes the issue without the need to cast back to float. Since I believe this is a better solution, I will update this PR. Thanks!

Reply:
excellent ! Glad to have nudged in a more optimal direction ;)